src/Traits/VersionNumberCacheTrait.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace SilverStripe\Versioned\Traits;
+
+use InvalidArgumentException;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DB;
+use SilverStripe\Versioned\ReadingMode;
+use SilverStripe\Versioned\Versioned;
+
+trait VersionNumberCacheTrait
+{
+    /**
+     * A cache used by get_versionnumber_by_stage().
+     * Clear through {@link flushCache()}.
+     * version (int)0 means not on this stage.
+     *
+     * @var array
+     */
+    protected static $cache_versionnumber;
+
+    /**
+     * Used to enable or disable the prepopulation of the version number cache.
+     * Defaults to true.
+     *
+     * @config
+     * @var boolean
+     */
+    private static $prepopulate_versionnumber_cache = true;
+
+    /**
+     * Shorthand method to call prepopulateVersionNumberCacheForStage() for each applicable stage
+     * This is cache the Version numbers of a particular DataClass
```

Comment:
```suggestion
     * This caches the version numbers of a particular DataObject class
```
I assume that's what you meant?

Reply:
Updated